diff --git a/data/mapObjects/silphco5.asm b/data/mapObjects/silphco5.asm
index 62e30fac..7cb395b2 100755
--- a/data/mapObjects/silphco5.asm
+++ b/data/mapObjects/silphco5.asm
@@ -20,7 +20,7 @@ SilphCo5Object:
 	object SPRITE_ROCKET, $1c, $4, STAY, UP, $5, OPP_ROCKET, $1d
 	object SPRITE_BALL, $2, $d, STAY, NONE, $6, TM_09
 	object SPRITE_BALL, $4, $6, STAY, NONE, $7, PROTEIN
-	object SPRITE_BALL, $15, $10, STAY, NONE, $8, CARD_KEY
+	object SPRITE_BALL, $15, $10, STAY, NONE, $8, $aa
 	object SPRITE_CLIPBOARD, $16, $c, STAY, NONE, $9 ; person
 	object SPRITE_CLIPBOARD, $19, $a, STAY, NONE, $a ; person
 	object SPRITE_CLIPBOARD, $18, $6, STAY, NONE, $b ; person
