diff --git a/data/hide_show_data.asm b/data/hide_show_data.asm
index 1fa4b64a..2b67e26d 100755
--- a/data/hide_show_data.asm
+++ b/data/hide_show_data.asm
@@ -299,8 +299,8 @@ MapHS0A:
 	db SAFFRON_CITY,$0B,Hide
 	db SAFFRON_CITY,$0C,Hide
 	db SAFFRON_CITY,$0D,Hide
-	db SAFFRON_CITY,$0E,Show
-	db SAFFRON_CITY,$0F,Hide
+	db SAFFRON_CITY,$0E,Hide
+	db SAFFRON_CITY,$0F,Show
 MapHS0D:
 	db ROUTE_2,$01,Show
 	db ROUTE_2,$02,Show
diff --git a/engine/items/items.asm b/engine/items/items.asm
index c4c1634b..b035eb9b 100755
--- a/engine/items/items.asm
+++ b/engine/items/items.asm
@@ -63,7 +63,7 @@ ItemUsePtrTable:
 	dw UnusableItem      ; BIKE_VOUCHER
 	dw ItemUseXAccuracy  ; X_ACCURACY
 	dw ItemUseEvoStone   ; LEAF_STONE
-	dw ItemUseCardKey    ; CARD_KEY
+	dw UnusableItem      ; CARD_KEY
 	dw UnusableItem      ; NUGGET
 	dw UnusableItem      ; ??? PP_UP
 	dw ItemUsePokedoll   ; POKE_DOLL
@@ -1549,75 +1549,31 @@ ItemUseXAccuracy:
 	set UsingXAccuracy,[hl] ; X Accuracy bit
 	jp PrintItemUseTextAndRemoveItem
 
-; This function is bugged and never works. It always jumps to ItemUseNotTime.
-; The Card Key is handled in a different way.
-ItemUseCardKey:
-	xor a
-	ld [wUnusedD71F],a
-	call GetTileAndCoordsInFrontOfPlayer
-	ld a,[GetTileAndCoordsInFrontOfPlayer]
-	cp a,$18
-	jr nz,.next0
-	ld hl,CardKeyTable1
-	jr .next1
-.next0
-	cp a,$24
-	jr nz,.next2
-	ld hl,CardKeyTable2
-	jr .next1
-.next2
-	cp a,$5e
-	jp nz,ItemUseNotTime
-	ld hl,CardKeyTable3
-.next1
+; Called from Poké Doll code to give special message against the ghost Marowak.
+; Returns set zero flag (and hl pointing to the message) if so.
+CheckWhetherPokedollingMarowak:
 	ld a,[wCurMap]
-	ld b,a
-.loop
-	ld a,[hli]
-	cp a,$ff
-	jp z,ItemUseNotTime
-	cp b
-	jr nz,.nextEntry1
-	ld a,[hli]
-	cp d
-	jr nz,.nextEntry2
-	ld a,[hli]
-	cp e
-	jr nz,.nextEntry3
-	ld a,[hl]
-	ld [wUnusedD71F],a
-	jr .done
-.nextEntry1
-	inc hl
-.nextEntry2
-	inc hl
-.nextEntry3
-	inc hl
-	jr .loop
-.done
-	ld hl,ItemUseText00
-	call PrintText
-	ld hl,wd728
-	set 7,[hl]
+	cp a,POKEMONTOWER_6
+	ret nz
+	ld a,[wEnemyMonSpecies2]
+	cp a,MAROWAK
+	ret nz
+	ld hl,NoPokedollingMarowakText
 	ret
 
-; These tables are probably supposed to be door locations in Silph Co.,
-; but they are unused.
-; The reason there are 3 tables is unknown.
+NoPokedollingMarowakText:
+	text "No! There are"
+	line "enough sequence"
+	cont "breaks in this"
+	cont "game already -"
+	cont "let's not add"
+	cont "any more!"
+	prompt
+	db "@"
 
-; Format:
-; 00: Map ID
-; 01: Y
-; 02: X
-; 03: ID?
-
-CardKeyTable1:
-	db  SILPH_CO_2F,$04,$04,$00
-	db  SILPH_CO_2F,$04,$05,$01
-	db  SILPH_CO_4F,$0C,$04,$02
-	db  SILPH_CO_4F,$0C,$05,$03
-	db  SILPH_CO_7F,$06,$0A,$04
-	db  SILPH_CO_7F,$06,$0B,$05
+; Remnants of the unused ItemUseCardKey tables.
+
+	db              $06,$0B,$05
 	db  SILPH_CO_9F,$04,$12,$06
 	db  SILPH_CO_9F,$04,$13,$07
 	db SILPH_CO_10F,$08,$0A,$08
@@ -1638,14 +1594,14 @@ CardKeyTable2:
 	db $ff
 
 CardKeyTable3:
-	db SILPH_CO_11F,$08,$09,$14
-	db SILPH_CO_11F,$09,$09,$15
-	db $ff
+	db SILPH_CO_11F,$08,$09
 
 ItemUsePokedoll:
 	ld a,[wIsInBattle]
 	dec a
 	jp nz,ItemUseNotTime
+	call CheckWhetherPokedollingMarowak
+	jp z, ItemUseFailed
 	ld a,$01
 	ld [wEscapedFromBattle],a
 	jp PrintItemUseTextAndRemoveItem
diff --git a/scripts/lavenderhouse1.asm b/scripts/lavenderhouse1.asm
index 91b9286b..3bbfe125 100755
--- a/scripts/lavenderhouse1.asm
+++ b/scripts/lavenderhouse1.asm
@@ -72,7 +72,7 @@ LavenderHouse1Text5:
 	jr nz, .asm_15ac2
 	ld hl, LavenderHouse1Text_1d94c
 	call PrintText
-	lb bc, POKE_FLUTE, 1
+	lb bc, $aa, 1
 	call GiveItem
 	jr nc, .BagFull
 	ld hl, ReceivedFluteText
diff --git a/scripts/pokemontower7.asm b/scripts/pokemontower7.asm
index 88c62c53..5525df48 100755
--- a/scripts/pokemontower7.asm
+++ b/scripts/pokemontower7.asm
@@ -263,9 +263,13 @@ PokemonTower7FujiText:
 	ld a, HS_SAFFRON_CITY_E
 	ld [wMissableObjectIndex], a
 	predef HideObject
-	ld a, HS_SAFFRON_CITY_F
+	; Don't show this guard - he's already shown.
+	; Otherwise we resurrect him if Tower is done after Silph.
+	jr .skipShowingSilphGuard
+	;ld a, HS_SAFFRON_CITY_F
 	ld [wMissableObjectIndex], a
 	predef ShowObject
+.skipShowingSilphGuard
 	ld a, $4
 	ld [wPokemonTower7CurScript], a
 	ld [wCurMapScript], a
diff --git a/text/maps/fujis_house.asm b/text/maps/fujis_house.asm
index 86cade68..3799e2b4 100644
--- a/text/maps/fujis_house.asm
+++ b/text/maps/fujis_house.asm
@@ -47,9 +47,11 @@ _LavenderHouse1Text_1d94c::
 
 _ReceivedFluteText::
 	text "<PLAYER> received"
-	line "a @"
+	line "@"
 	TX_RAM wcf4b
 	text "!@@"
+	nop
+	nop
 
 _FluteExplanationText::
 	db $0
