diff --git a/data/mapObjects/rockethideout4.asm b/data/mapObjects/rockethideout4.asm
index b4e98d04..c504225d 100755
--- a/data/mapObjects/rockethideout4.asm
+++ b/data/mapObjects/rockethideout4.asm
@@ -16,7 +16,7 @@ RocketHideout4Object:
 	object SPRITE_BALL, $a, $c, STAY, NONE, $5, HP_UP
 	object SPRITE_BALL, $9, $4, STAY, NONE, $6, TM_02
 	object SPRITE_BALL, $c, $14, STAY, NONE, $7, IRON
-	object SPRITE_BALL, $19, $2, STAY, NONE, $8, SILPH_SCOPE
+	object SPRITE_BALL, $19, $2, STAY, NONE, $8, $aa
 	object SPRITE_BALL, $a, $2, STAY, NONE, $9, LIFT_KEY
 
 	; warp-to
